dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/fields/FieldsPathParamsHelper.java

```diff
@@ -42,35 +41,30 @@
  * based on {@link FieldFilterParser }. It follows the principles of
  * {@link org.hisp.dhis.fieldfiltering.FieldFilterService}
  */
-public interface InstanceFieldsParamMapper
+public class FieldsPathParamsHelper
 {
-    String FIELD_RELATIONSHIPS = "relationships";
+    private FieldsPathParamsHelper()
```

Comment:
I think its a good idea to prevent instantiation (by mistake) and document why you added a private constructor via `throw new IllegalStateException( "Utility class" );`.

Reply:
you can't instantiate the class unless you change the visibility so I don't see the benefit to add that

Reply:
Joshua Bloch explains it in Effective Java as:

> the exception isn’t strictly required, but it provides insurance in case the constructor is accidentally invoked from within the class. It guarantees the class will never be instantiated under any circumstances.

The message `"Utility class"` also acts as a documentation of intent. We also use this idiom a lot, so it would simply be consistent.